### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Layrovell**
[DEMO LINK](https://Layrovell.github.io/react_tabs/)

### CODE REVIEW

**Review by Serhii-Alieksieievych — APPROVED** [review]

File: src/App.js [hunk 1]
```diff
@@ -1,24 +1,43 @@
...
 
-const App = () => (
-  <h1>React tabs</h1>
-);
+class App extends React.Component {
+  state = {
+    active: 'Tab 1',
+  }
+
+  handlerOnClick = (e) => {
```
Comment (aliceblunt8) [inline src/App.js:23]:
```suggestion
 selectTab = (e) => {
```

File: src/components/Tab/Tab.css [hunk 2]
```diff
@@ -0,0 +1,17 @@
+/*.container {*/
```
Comment (aliceblunt8) [inline src/components/Tab/Tab.css:1]: delete empty files

File: src/components/Tab/Tab.jsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import './Tab.css';
+import PropTypes from 'prop-types';
+
+export const Tab = ({ tabs, active, handlerOnClick }) => {
+
+  return (
```
Comment (aliceblunt8) [inline src/components/Tab/Tab.jsx:7]:
```suggestion
export const Tab = ({ tabs, active, handlerOnClick }) => (
```

File: src/components/Tab/Tab.jsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+          }}
+          className={active === tab.title ? 'active' : ''}
+        >
+          {tab.title}
+        </button>
+      ))}
+      <div className="content">
+        {tabs.filter(tab => tab.title === active)
+          .map(tab => <p>{tab.content}</p>
+          )}
```
Comment (aliceblunt8) [inline src/components/Tab/Tab.jsx:23]: render this in App

File: src/App.js [hunk 5]
```diff
@@ -1,24 +1,43 @@
...
+    title: 'Tab 3', content: 'Some text 3',
   },
 ];
 
-const App = () => (
-  <h1>React tabs</h1>
-);
+class App extends React.Component {
+  state = {
+    active: 'Tab 1',
```
Comment (aliceblunt8) [inline src/App.js:20]: save selected tab in here

File: src/components/Tab/Tab.jsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+export const Tab = ({ tabs, active, handlerOnClick }) => {
+
+  return (
+    <div className="container">
+      {tabs.map(tab => (
+        <button
+          type="button"
+          onClick={() => {
+            handlerOnClick(tab.title);
+          }}
```
Comment (aliceblunt8) [inline src/components/Tab/Tab.jsx:14]:
```suggestion
          onClick={() =>handlerOnClick(tab.title)}
```

**Review by aliceblunt8 — CHANGES_REQUESTED** [review]

File: src/App.js [hunk 7]
```diff
@@ -1,24 +1,46 @@
...
+  };
+
+  render() {
+    return (
+      <div className="App">
+        <h1>React tabs</h1>
+        <Tab
+          tabs={tabs}
+          active={this.state.active}
+          handlerOnClick={this.selectTab}
```
Comment (aliceblunt8) [inline src/App.js:35]:
```suggestion
          selectTab={this.selectTab}
```

File: src/App.js [hunk 8]
```diff
@@ -1,24 +1,46 @@
...
+    title: 'Tab 3', content: 'Some text 3',
   },
 ];
 
-const App = () => (
-  <h1>React tabs</h1>
-);
+class App extends React.Component {
+  state = {
+    active: this.selectTab,
```
Comment (aliceblunt8) [inline src/App.js:19]: I meant not this)) now you save object method in state, but I meant to save active tab (object) or it's index (number) in App's state

File: src/App.js [hunk 9]
```diff
@@ -1,24 +1,46 @@
...
+      <div className="App">
+        <h1>React tabs</h1>
+        <Tab
+          tabs={tabs}
+          active={this.state.active}
+          handlerOnClick={this.selectTab}
+        />
+        <div className="content">
+          {tabs.filter(tab => tab.title === this.state.active)
+            .map(tab => <p>{tab.content}</p>)}
```
Comment (aliceblunt8) [inline src/App.js:39]: if you save selected tab's index in state you won't need to filter tabs in order to find one
you can just `tabs[this.state.selectedTabIndex].content`

File: src/components/Tab/Tab.jsx [hunk 10]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export const Tab = ({ tabs, active, handlerOnClick }) => (
+  <div className="container">
+    {tabs.map(tab => (
+      <button
+        type="button"
+        onClick={() => handlerOnClick(tab.title)}
```
Comment (aliceblunt8) [inline src/components/Tab/Tab.jsx:9]: I think it's better to pass index, not title

**Review by aliceblunt8 — CHANGES_REQUESTED** [review]

File: src/App.js [hunk 11]
```diff
@@ -1,24 +1,45 @@
...
+    title: 'Tab 3', content: 'Some text 3',
   },
 ];
 
-const App = () => (
-  <h1>React tabs</h1>
-);
+class App extends React.Component {
+  state = {
+    active: 0,
```
Comment (aliceblunt8) [inline src/App.js:19]:
```suggestion
    activeTabId: 0,
```

File: src/components/Tab/Tab.jsx [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+import PropTypes from 'prop-types';
+
+export const Tab = ({ tabs, active, selectTab }) => (
+  <div className="container">
+    {tabs.map((tab, idx) => (
+      <button
+        type="button"
+        value={idx}
+        onClick={() => selectTab(idx)}
+        className={active === idx ? 'active' : ''}
```
Comment (aliceblunt8) [inline src/components/Tab/Tab.jsx:11]: you can use classnames lib
https://www.npmjs.com/package/classnames

**Review by aliceblunt8 — APPROVED** [review]

**Pull request closed without merging**